src/components/Projects/Current/StatsAllProjects.tsx

```diff
@@ -25,11 +26,16 @@ export default function StatsAllProjects({ projects }: Props) {
     return finishAt.isAfter(Time()) && finishAt.isBefore(Time().add(1, 'week'))
   })
 
-  const currentQuarter = Time().quarter()
+  const { year: yearParam, quarter: quarterParam } = useYearAndQuarter()
```

Comment:
If `useYearAndQuarter` already checks for

```typescript
 const year = validateYear(yearParam) ?? Time().year()
  const quarter = validateQuarter(quarterParam) ?? Time().quarter()
  ``` 
  
  then you can avoid this part right?
  
  ```typescript
  const isYearAndQuarterValid = yearParam && quarterParam

  const currentYear = isYearAndQuarterValid ? yearParam : Time().year()
  const currentQuarter = isYearAndQuarterValid ? quarterParam : Time().quarter()
  ```

Reply:
You could also return an `isValid` property from the hook as you are also doing that check in another component.